requests/models.py

```diff
@@ -470,18 +469,12 @@ def prepare_body(self, data, files, json=None):
         self.body = body
 
     def prepare_content_length(self, body):
-        if hasattr(body, 'seek') and hasattr(body, 'tell'):
-            curr_pos = body.tell()
-            body.seek(0, 2)
-            end_pos = body.tell()
-            self.headers['Content-Length'] = builtin_str(max(0, end_pos - curr_pos))
-            body.seek(curr_pos, 0)
-        elif body is not None:
-            l = super_len(body)
-            if l:
-                self.headers['Content-Length'] = builtin_str(l)
-        elif (self.method not in ('GET', 'HEAD')) and (self.headers.get('Content-Length') is None):
-            self.headers['Content-Length'] = '0'
+        if self.headers.get('Content-Length') is None:
+            if body is not None:
+                length = super_len(body)
+                self.headers['Content-Length'] = builtin_str(length)
+            elif self.method not in ('GET', 'HEAD'):
+                self.headers['Content-Length'] = '0'
```

Comment:
First note: we definitely need to comment this better to explain what we're doing. In this case, we're saying that we only want to override the `Content-Length` header if it hasn't already been set for any reason.

My question is: are we sure that that's the logic we want? Your description suggests that `prepare_content_length` should have an `overwrite` param, but this code doesn't add it.

Can you elaborate on what's going on here?